### ISSUE TITLE
AGN torus SED drops 2700x between two adjacent grid points at exactly 1 mm

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#1512]
## What

With a SKIRTOR torus, `state.derived["sed_agn"]` falls by a factor **2700 between two
adjacent panchromatic grid points** at exactly 1e7 A (1 mm), preceded by an accelerating
plunge. Measured in **float64**, on `main`'s physics — this is not a float32 issue.

```
 wave [A]        sed_agn        ratio to previous point
 9.727e+06      2.23654e+29        0.7577
 9.817e+06      1.50626e+29        0.6735
 9.908e+06      7.60894e+28        0.5052
 1.000e+07      2.80334e+25        0.0003684   <-- cutoff
 1.009e+07      2.71752e+25        0.9694
 1.019e+07      2.62606e+25        0.9663
```

Grid spacing here is ~0.9% in wavelength. A Rayleigh-Jeans tail over such a step gives a
ratio of ~0.982. The observed 0.505 (and then 3.7e-4) is not a physical SED shape.

## It is the torus, and it is not in the data

* **It is the torus.** Rebuilding the same model without the torus block gives a perfectly
  smooth SED across 1 mm — ratio 1.009 at every step.
* **It is not in the template.** SKIRTOR's native grid has a node at exactly 1e7 A
  (index 131 of 136) and is a clean power law straight through it: the log-log slope is
  **-5.90 on both sides** (the larger raw flux drop after the node is only because node
  spacing widens from x1.148 to x2 there). No template has a zero or negative flux at any
  of the last six nodes.
* **It is not new.** The same cutoff is present, to within rounding, at commits predating
  #1485 (ratio 3.448e-4 vs 3.684e-4), so the log-space resampling change did not cause it.

So the model introduces a discontinuity that the underlying library does not have.

## Why it matters

1. **Possible physics error in the mm band.** If the cutoff is unintended, AGN torus flux
   is wrong by orders of magnitude beyond 1 mm, which is inside the range panchromatic
   (AGN + radio) fits actually use.
2. **It makes precision comparisons ill-posed.** Two dtypes can land on opposite sides of
   a boundary one ulp wide, so the measured "error" is the size of the step. This is how it
   surfaced: `test_disc_is_exact_in_float32[powerlaw]` reported `max_rel=3.31e-02` at that
   single point while every other point in the SED was within 7e-06.

## First question to settle

Is the cutoff **deliberate** — e.g. a handoff where the radio component takes over beyond
1 mm — or an accident of how the torus is composed onto the panchromatic grid? If deliberate,
it should be tapered rather than a hard step, and documented; if not, it is a flux bug.

I have not changed any physics here. The float32 test was narrowed to skip points where the
float64 reference is discontinuous (mask derived from the reference only, with a budget so it
cannot empty the comparison), which is correct independent of how this issue resolves.

## Reproduce

Build stellar + two-component dust + `agn={'type':'composable', 'disc':{'type':'powerlaw'},
'torus':{'type':'skirtor'}, 'norm':'cigale_joint', 'fracAGN':0.1}` at `redshift=0.1` under
`jax.enable_x64(True)`, then inspect `predict_state(p).derived["sed_agn"]` around index of
`wavelengths == 1e7`.

**Comment by suchethac** [comment]
Reproduced on main 8e642ab11 (float64, tracked SSP `ssp_prsc_miles_chabrier_wNE_logGasU-3.0_logGasZ0.0.h5`), and it is larger than the 2700 reported. Composable `agn={'type': 'composable', 'disc': {'type': 'powerlaw'}, 'torus': {'type': 'skirtor'}, 'norm': 'cigale_joint', 'fracAGN': 0.1}` at z = 0.1, `state.derived["sed_agn"]` on the model's 6127-point grid:

| wavelength [Å] | sed_agn | ratio to previous |
|---|---|---|
| 9.727e6 | 2.37e29 | 0.758 |
| 9.817e6 | 1.60e29 | 0.673 |
| 9.908e6 | 8.07e28 | 0.505 |
| 1.000e7 | 2.64e23 | 3.3e-6 |
| 1.009e7 | 2.55e23 | 0.966 |

Controls on the same build: without the torus block, ratio 1.009 at every step; monolithic `agn={'type': 'skirtor'}`, ratio 1.009 at every step. The SKIRTOR template itself has a native node at exactly 1e7 Å and runs to 1e8 Å, so the collapse is introduced by the composable torus composition, with the accelerating decline over the three points before 1e7 Å pointing at interpolation toward a zero node. A fix lane is localizing the zero and will open a PR closing this issue.
